src/CommunityToolkit.Maui.SourceGenerators/Generators/TextColorToGenerator.cs

```diff
@@ -0,0 +1,158 @@
+
+		// Get Microsoft.Maui.Controls Assymbly Symbol
+		var mauiControlsAssemblySymbolProvider = context.CompilationProvider.Select(
+			static (compilation, token) => compilation.SourceModule.ReferencedAssemblySymbols.Single(q => q.Name == mauiControlsAssembly));
+
+		var inputs = userGeneratedClassesProvider.Collect()
+						.Combine(mauiControlsAssemblySymbolProvider)
+						.Select(static (combined, cancellationToken) => (UserGeneratedClassesProvider: combined.Left, MauiControlsAssemblySymbolProvider: combined.Right))
+						.Combine(context.CompilationProvider)
+						.Select(static (combined, cancellationToken) => (combined.Left.UserGeneratedClassesProvider, combined.Left.MauiControlsAssemblySymbolProvider, Compilation: combined.Right));
+
+		context.RegisterSourceOutput(inputs, (context, collectedValues) =>
+		Execute(context, collectedValues.Compilation, collectedValues.UserGeneratedClassesProvider, collectedValues.MauiControlsAssemblySymbolProvider));
+	}
+
+	static void Execute(SourceProductionContext context, Compilation compilation, ImmutableArray<ClassDeclarationSyntax> userGeneratedClassesProvider, IAssemblySymbol mauiControlsAssemblySymbolProvider)
+	{
+		var textStyleSymbol = compilation.GetTypeByMetadataName(iTextStyleInterface);
+		var iAnimatableSymbol = compilation.GetTypeByMetadataName(iAnimatableInterface);
+
+		if (textStyleSymbol is null || iAnimatableSymbol is null)
+			return;
+
+		var textStyleClassList = new List<(string ClassName, string ClassAcessModifier, string Namespace)>();
+
+		// Collect Microsoft.Maui.Controls that Implement ITextStyle
+		var mauiTextStyleImplementors = mauiControlsAssemblySymbolProvider.GlobalNamespace.GetNamedTypeSymbols().Where(x => x.AllInterfaces.Contains(textStyleSymbol, SymbolEqualityComparer.Default)
+				&& x.AllInterfaces.Contains(iAnimatableSymbol, SymbolEqualityComparer.Default));
+
+		foreach (var namedTypeSymbol in mauiTextStyleImplementors)
+			textStyleClassList.Add((namedTypeSymbol.Name, "public", namedTypeSymbol.ContainingNamespace.ToDisplayString()));
+
+		// Collect All Classes in User Library that Implement ITextStyle
+		foreach (var classDeclarationSyntax in userGeneratedClassesProvider)
+		{
+			var declarationSymbol = compilation.GetSymbol<INamedTypeSymbol>(classDeclarationSyntax);
+			if (declarationSymbol is null)
+				continue;
+			if (declarationSymbol.AllInterfaces.Contains(textStyleSymbol, SymbolEqualityComparer.Default)
+				&& declarationSymbol.AllInterfaces.Contains(iAnimatableSymbol, SymbolEqualityComparer.Default))
+			{
+				textStyleClassList.Add((declarationSymbol.Name, GetClassAccessModifier(declarationSymbol), declarationSymbol.ContainingNamespace.ToDisplayString()));
+			}
+		}
+
+		var options = ((CSharpCompilation)compilation).SyntaxTrees[0].Options as CSharpParseOptions;
+		foreach (var textStyleClass in textStyleClassList)
+		{
+			var textColorToBuilder = new StringBuilder(@"
+// AutoGenerated Code
```

Comment:
If we can't change the `.shared.cs` suffix, the compiler also recognizes a comment on this form.

```suggestion
// <auto-generated/>
```